packages/block-library/src/query-pagination-numbers/index.php

```diff
@@ -45,6 +45,22 @@ function render_block_core_query_pagination_numbers( $attributes, $content, $blo
 			'total'     => $total,
 			'prev_next' => false,
 		);
+		if ( 1 !== $page ) {
+			/**
+			 * `paginate_links` doesn't use the provided `format` when the page is `1`.
+			 * This is great for the main query as it removes the extra query params
+			 * making the URL shorter, but in the case of multiple custom queries is
+			 * problematic. It results in returning an empty link which ends up with
+			 * a link to the current page.
+			 *
+			 * A way to address this is to add a `fake` query arg with no value that
+			 * is the same for all custom queries. This way the link is not empty and
+			 * preserves all the other existent query args.
+			 *
+			 * @see https://developer.wordpress.org/reference/functions/paginate_links/
+			 */
+			$paginate_args['add_args'] = array( 'cst' => '' );
```

Comment:
What does `cst` stand for here, is it just a random useless param?

Reply:
I feel ideally, we should add a dedicated argument/option in `paginate_links` to disable the default behavior of skipping `format`

Reply:
So this causes `?cst` to be added to every link? 🤔  Seems a bit unexpected/weird.

Why not just filter `paginate_links_output`?

Theoretically it could cause some confusion if someone's already using a query arg with this name and expects it to do something else.

Could you perhaps share what the expected output is with this change?

Agreed that this should be fixed with a patch in core.

Reply:
>What does cst stand for here, is it just a random useless param?

Yes.

>Theoretically it could cause some confusion if someone's already using a query arg with this name 

We can change this to whatever. That's why I didn't used a single character, as it would increase those chances. I'm open to any suggestions.

>Why not just filter paginate_links_output?

This would require parsing the html string and I'm not sure it worths it and how complicated it can get with other filters having been applied before, to other parts of this html construction. An extra variable in core should be the best solution probably.